### DISCUSSION TITLE
font-lock files in git-commit-mode

### DISCUSSION THREAD

**Discussion by haze** [magit/magit#5101] [Ideas]
It would be super cool if `git-commit-mode` could apply the syntax highlighting from the derived mode from the file type (if applicable) for verbose commits (`git commit —verbose`)

**Comment by tarsius** [comment]
Thanks for the suggestion. It's unlikely that I will work on this. IMO this too minor an issue to warrant the effort. Personally I have disabled this diff altogether and instead rely on the diff shown in the separate buffer, and I expect that many other users have done the same.

Diffs that are displayed in actual Magit buffers probably will gain syntax highlighting at some point, but that too isn't a priority. Not much has changed regarding this, since I wrote https://github.com/magit/magit/issues/2942#issuecomment-933440063.

**Reply by haze** [reply]
How can I see the diff in a separate buffer? I don't initiate a commit from emacs, so if it's obvious, I haven't ran into it yet. I just use emacs as the git editor when running commit from the command line.

**Reply by kyleam** [reply]
Call `magit-diff-while-committing` (bound by default to `C-c C-d`).
